File: xadditus-android/app/src/main/java/com/linoagli/android/xadditus/utils/SystemInfo.java
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.utils
 *
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.utils;

import java.util.regex.Pattern;

public class SystemInfo {
    public static final String SYSTEM_PROPERTY_OS_NAME = "os.name";
    public static final String SYSTEM_PROPERTY_OS_VERSION = "os.version";
    public static final String SYSTEM_PROPERTY_HOME_DIR = "user.home";
    public static final String SYSTEM_PROPERTY_CURRENT_WORKING_DIR = "user.dir";

    public enum OperatingSystem { Windows, Linux, SunOS, FreeBSD, Mac, None }

    public static String get(String systemProperty)
    {
        return System.getProperty(systemProperty);
    }

    public static OperatingSystem getOperatingSystemFromName(String osName)
    {
        osName = osName.toLowerCase();

        for (OperatingSystem os : OperatingSystem.values())
        {
            Pattern pattern = Pattern.compile(os.name().toLowerCase());
            boolean found = pattern.matcher(osName).find();

            if (found) return os;
        }

        return OperatingSystem.None;
    }

    public static boolean isLinux()
    {
        String osName = get(SYSTEM_PROPERTY_OS_NAME);
        OperatingSystem os = getOperatingSystemFromName(osName);

        return os == OperatingSystem.Linux;
    }

    public static boolean isWindows()
    {
        String osName = get(SYSTEM_PROPERTY_OS_NAME);
        OperatingSystem os = getOperatingSystemFromName(osName);

        return os == OperatingSystem.Windows;
    }
}

File: xadditus-android/app/src/main/kotlin/com/linoagli/android/xadditus/helpers/SystemInfo.kt
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.helpers
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.helpers

import java.util.regex.Pattern

object SystemInfo {
    val SYSTEM_PROPERTY_OS_NAME = "os.name"
    val SYSTEM_PROPERTY_OS_VERSION = "os.version"
    val SYSTEM_PROPERTY_HOME_DIR = "user.home"
    val SYSTEM_PROPERTY_CURRENT_WORKING_DIR = "user.dir"

    enum class OperatingSystem {
        Windows, Linux, SunOS, FreeBSD, Mac, None
    }

    operator fun get(systemProperty: String): String {
        return System.getProperty(systemProperty)
    }

    fun getOperatingSystemFromName(osName: String): OperatingSystem {
        var osName = osName
        osName = osName.toLowerCase()

        for (os in OperatingSystem.values()) {
            val pattern = Pattern.compile(os.name.toLowerCase())
            val found = pattern.matcher(osName).find()

            if (found) return os
        }

        return OperatingSystem.None
    }

    fun isLinux(): Boolean {
        val osName = get(SYSTEM_PROPERTY_OS_NAME)
        val os = getOperatingSystemFromName(osName)
        return os == OperatingSystem.Linux
    }

    fun isWindows(): Boolean {
        val osName = get(SYSTEM_PROPERTY_OS_NAME)
        val os = getOperatingSystemFromName(osName)
        return os == OperatingSystem.Windows
    }
}